From 4f32f36559dea3b3da341c1c574babfbbbfab1f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Fri, 19 Sep 2025 15:07:00 +0200
Subject: [PATCH 1/2] compositor: Destroy window drag after other 'end'
 handlers

This means other handlers will have had enough time to cleanup their
references to the window drag, before it gets finalized.

Closes: https://gitlab.gnome.org/GNOME/mutter/-/issues/4324
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4675>
---
 src/compositor/compositor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/compositor/compositor.c b/src/compositor/compositor.c
index 79db7d2909..146a52653e 100644
--- a/src/compositor/compositor.c
+++ b/src/compositor/compositor.c
@@ -1602,8 +1602,8 @@ meta_compositor_drag_window (MetaCompositor      *compositor,
       return FALSE;
     }
 
-  g_signal_connect (priv->current_drag, "ended",
-                    G_CALLBACK (on_window_drag_ended), compositor);
+  g_signal_connect_after (priv->current_drag, "ended",
+                          G_CALLBACK (on_window_drag_ended), compositor);
   return TRUE;
 }
 
-- 
2.51.0

